susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+    if run_command(command):
+        print("PostgreSQL %sed" % action)
+    else:
+        raise RuntimeError("Could not %s PostgreSQL!" % action)
+
+
+def dump_database(dumpName, excludedTables=[]):
+    dbName = None
+    with open(sys.path[0] + '/clear-db-answers-pgsql.txt') as f:
+        for line in f.read().splitlines():
+            if line.split('=')[0] == "db-name":
+                dbName = line.split('=')[1]
+    command = "/usr/bin/su - postgres -c 'pg_dump --dbname=%s --file=%s" % (
+        dbName, dumpName)
+    for excludedTable in excludedTables:
+        command += " -T %s" % excludedTable
+    command += "'"
+    if run_command(command):
+        print("%s dumped correctly to %s" % (dbName, dumpName))
+    else:
+        raise RuntimeError("Could not dump %s!" % dbName)
+
+
+def run_upgrade(upgradeScript, schemaPath, newVersion):
+    logPath = "/var/log/spacewalk/schema-upgrade/"
+    command = "export SUMA_TEST_SCHEMA_VERSION=%s; %s -y" % (
+        newVersion, upgradeScript)
+    if run_command(command):
+        print("Upgrade executed successfully:")
+    else:
+        for element in os.listdir(logPath):
+            if re.search('^schema-from-.*\.log$', element):
+                print("Content of file %s/%s" % (logPath, element))
+                with open(logPath + '/' + element) as f:
+                    for line in f.read().splitlines():
+                        print(line)
+        raise RuntimeError(
+            "Upgrade failed! Most probably the new SQL scripts are not idempotent!")
+
+
+def diff_dumps(initial_dump, migrated_dump):
+    fileInitial = open(initial_dump)
+    fileMigrated = open(migrated_dump)
+    return(difflib.context_diff(fileInitial.readlines(), fileMigrated.readlines()))
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Test idempotency of SQL scripts")
+    parser.add_argument("-s", "--schema-path", action="store", dest="schemaPath",
+                        help="Path where the directories with the schema upgrades are (default: /etc/sysconfig/rhn/schema-upgrade",
+                        default="/etc/sysconfig/rhn/schema-upgrade")
+    parser.add_argument("-p", "--from-pr", action="store", dest="prFile",
+                        help="Check files with changes from a PR by rerunning added or changed files. The value is the path to a JSON file with PR information generated by gitarro")
+    parser.add_argument("-v", "--from-version", action="store", dest="version",
+                        help="Check all files since a base version (for example 4.0.0)")
+    parser.add_argument("-u", "--upgrade-script", action="store", dest="upgradeScript",
+                        help="Path where the spacewalk-schema-upgrade script is (default: /manager/schema/spacewalk/spacewalk-schema-upgrade)",
+                        default="/manager/schema/spacewalk/spacewalk-schema-upgrade")
+    args = parser.parse_args()
+    if args.schemaPath is None:
```

Comment:
i would create a `argparser` helper function for extracting this from main

Reply:
I will give it a try tomorrow.